packages/opentelemetry-instrumentation/README.md

```diff
@@ -0,0 +1,153 @@
+[![devDependencies][devDependencies-image]][devDependencies-url]
+[![Apache License][license-image]][license-image]
+
+## Installation
+
+```bash
+npm install --save @opentelemetry/instrumentation
+```
+
+## Usage in Node
+
+```typescript
+import {
+  Instrumentation,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+} from '@opentelemetry/instrumentation';
+
+import * as module_name_to_be_pached from 'module_name_to_be_pached';
+
+export class MyPlugin extends Instrumentation {
+  constructor(config: api.InstrumentationConfig = {}) {
+    super('MyPlugin', VERSION, config);
+  }
+
+  // init method is default method that needs to be defined where
+  // definitions for patching should be defined. It will be then called
+  // before enabling the plugin and loading
+  protected _init() {
+    const module = new InstrumentationNodeModuleDefinition<typeof module_name_to_be_pached>(
+      'module_name_to_be_pached',
+      ['1.*'],
+       (moduleExports: typeof module_name_to_be_pached) => {
+        this._wrap(
+          moduleExports,
+          'mainMethodName',
+          this._patchMainMethodName()
+        );
+       },
+       (moduleExports: typeof module_name_to_be_pached) => {
+        this._unwrap(moduleExports, 'mainMethodName');
+       }
+    );
+    // in case you need to patch additional files - this is optional
+    module.files.push(this._addPatchingMethod());
+
+    return module;
+    // you can also define more modules then just return an array of modules
+    // return [module1, module2, ....]
+  }
+
+  private _addPatchingMethod(): InstrumentationNodeModuleFile<typeof module_name_to_be_pached> {
+    const file = new InstrumentationNodeModuleFile<typeof module_name_to_be_pached>(
+      'module_name_to_be_pached/src/some_file.js',
+      (moduleExports: typeof module_name_to_be_pached) => {
+        this._wrap(
+          moduleExports,
+          'methodName',
+          this._patchMethodName()
+        );
+        return moduleExports;
+      },
+      (moduleExports: typeof module_name_to_be_pached) => {
+        this._unwrap(moduleExports, 'methodName');
+      }
+    );
+    return file;
+  }
+
+  private _patchMethodName(): (original) => any {
+    const plugin = this;
+    return function methodName(original) {
+      return function patchMethodName(this: any): PromiseOrValue<module_name_to_be_pached.methodName> {
+        console.log('methodName', arguments);
+        return original.apply(this, arguments);
+      };
+    };
+  }
+
+  private _patchMainMethodName(): (original) => any {
+    const plugin = this;
+    return function mainMethodName(original) {
+      return function patchMainMethodName(this: any): PromiseOrValue<module_name_to_be_pached.mainMethodName> {
+        console.log('mainMethodName', arguments);
+        return original.apply(this, arguments);
+      };
+    };
+  }
+}
+
+// Later
+
+const myPLugin = new MyPlugin();
+myPLugin.setTracerProvider(provider);
```

Comment:
Maybe add a comment that setting tracer and meter provider is optional and API will be used.